api/storage_service.cc

```diff
@@ -465,6 +465,24 @@ void unset_repair(http_context& ctx, routes& r) {
     ss::force_terminate_all_repair_sessions_new.unset(r);
 }
 
+static sstables_loader::stream_scope parse_stream_scope(std::string_view scope) {
+    if (scope.empty() || scope == "all") {
+        return sstables_loader::stream_scope::all;
+    }
+    if (scope == "dc") {
+        return sstables_loader::stream_scope::dc;
+    }
+    if (scope == "rack") {
+        return sstables_loader::stream_scope::rack;
+    }
+    if (scope == "node") {
+        return sstables_loader::stream_scope::node;
+    }
+
+    throw httpd::bad_param_exception("invalid scope parameter value");
+    return sstables_loader::stream_scope::all;
```

Comment:
clang does not blame us for not returning here, if we don't have this statement. as `throw` is inherently `[[noreturn]]`. see https://godbolt.org/z/xhrr3rh3e

Reply:
Thanks!